### DISCUSSION TITLE
Valetudo 2021.05.0

### DISCUSSION THREAD

**Discussion by Hypfer** [Hypfer/Valetudo#944] [Release Discussions]
<div align="center">
    <img src="https://github.com/Hypfer/Valetudo/blob/master/assets/logo/valetudo_logo_with_name.svg" width="800" alt="valetudo">
    <p align="center"><h2>2021.05.0</h2></p>
</div>

This release features mostly bugfixes, unfinished features and a hostile takeover.

## MQTT Rewrite

Due to the scale of the MQTT rewrite that happened with 2021.04.0, some new bugs were introduced, which have been fixed with this release.

Furthermore, this release removes the migration logic of the old mqtt config format so if you're planning on upgrading from something other than 2021.04.0, make sure to upgrade to that before installing 2021.05.0.

As always, reading all the release notes is strongly recommended during upgrades.

## Not-yet finished things

Not everything in this section is already part of this release.

### UI Rewrite

@jomik is still working on the rewrite of the Valetudo UI. It's already looking fantastic:

![image](https://user-images.githubusercontent.com/974410/119648987-635ae980-be22-11eb-82c9-3b1c999303c4.png)
![image](https://user-images.githubusercontent.com/974410/119649000-6655da00-be22-11eb-831a-cb816a3991be.png)

If you're a frontend dev, a design person etc., feel free to join in. :)
We'll definitely need some design input, custom icons and much more stuff.

### Timers

A simple scheduler feature has been added to the backend for setups where a full-blown home automation system isn't required.
These shall be understood as WIP and can currently only be controlled [via REST API calls](https://github.com/Hypfer/Valetudo/discussions/824#discussioncomment-761574).

### Dreame Support

Public root coming soon™, again.

The beta test has been expanded to more users. If you want to take part in that, make sure to join the [Telegram Dreame Usergroup](https://t.me/joinchat/2qGFUaQ2pV9hNzUy) and check out the pinned form.
Currently, the announcement of the Dreame W10 and Z10 is delaying the release of the rooting method.

## Freenode hostile takeover

As you might've heard, Freenode, the FOSS IRC network has been taken over by the crown prince of korea, who decided to have some fun with it.

Today, the #Valetudo Freenode channel was also taken over:

![image](https://user-images.githubusercontent.com/974410/119649150-97360f00-be22-11eb-9508-2dfb93ea39be.png)

There are other and much more popular victims of this:

- https://mastodon.sdf.org/@kline/106299403921451814
- https://news.ycombinator.com/item?id=27286628
- https://twitter.com/fosdem/status/1397454352835653632

I've left the network and strongly advice you to do the same.
[https://libera.chat/](https://libera.chat/) is the continuation of Freenode with a different name, but the same Team that has been running Freenode for the last 20 years.

Freenode on the other hand is just the same name but with completely different people.

Stick to the community. Not to the brand.

For more information, check out some of the resignation letters of the former freenode and current libera staff:

- kline: https://www.kline.sh/
- jess: https://gist.github.com/jesopo/45a3e9cdbe517dc55e6058eb43b00ed9
- Md: https://blog.bofh.it/debian/id_461
- niko: https://coevoet.fr/freenode.html
- edk: https://gist.github.com/edk0/478fb4351bc3ba458288d6878032669d
- emilsp: https://gist.github.com/pinkisemils/39d4ded8b1639b6f39dcab15618649f5
- mniip: https://mniip.com/freenode.txt
- Swant: https://swantzter.se/freenode-resignation/
- JonathanD: https://gist.github.com/JonathanD82/6518b93708e0aaf4b8f92c8e7200816d

There's also a neat FAQ by @joepie91: https://gist.github.com/joepie91/df80d8d36cd9d1bde46ba018af497409

### Features

* **vendor.dreame:** Add iteration count to MapSegmentationCapability ([05b338f](https://github.com/Hypfer/Valetudo/commit/05b338f069feb7c311dcc11d7b50d91e310e79d3))
* Timers ([7b8c37a](https://github.com/Hypfer/Valetudo/commit/7b8c37ad1006aec62e1f19e5aec83d54da94189d))
* **ntpClient:** Keep track of the current state and enable configuration via REST ([e350eba](https://github.com/Hypfer/Valetudo/commit/e350eba43ab36076d406c02f933c7be108cc8e99))
* **vendor.dreame:** 1C: Add/fix various Map & Volume capabilities ([#915](https://github.com/Hypfer/Valetudo/issues/915)) ([c88df12](https://github.com/Hypfer/Valetudo/commit/c88df12de41ec05797adeb4c5215bf4d18456c9b))
* **vendor.roborock:** Add support for ordered segment cleanup with multiple iterations ([2541113](https://github.com/Hypfer/Valetudo/commit/25411138688454d43026d07c12e46b5eab5e9269))
* **vendor.viomi:** Implement DND capability ([#877](https://github.com/Hypfer/Valetudo/issues/877)) ([dd05c51](https://github.com/Hypfer/Valetudo/commit/dd05c51f4dfa45ffdf973b95e6b4851e423a176c))

### Bug Fixes

* **miio:** Better map upload logline ([0e7ce7c](https://github.com/Hypfer/Valetudo/commit/0e7ce7c7486498c55dd0443dd9cb0251fde1dab2))
* **mqtt:** Allow strings as segment IDs ([#891](https://github.com/Hypfer/Valetudo/issues/891)) ([3e30414](https://github.com/Hypfer/Valetudo/commit/3e304146980c8dc2a394a54e7346528ec5a07620)), closes [#889](https://github.com/Hypfer/Valetudo/issues/889)
* **mqtt:** Always publish state ([6a041c6](https://github.com/Hypfer/Valetudo/commit/6a041c62cfff4d2b04eb48e15ee20373512838fe))
* **mqtt:** Only migrate HAss topics if enabled ([#912](https://github.com/Hypfer/Valetudo/issues/912)) ([eb2edaf](https://github.com/Hypfer/Valetudo/commit/eb2edaf969a175648cb0f53405b1780f0ac799d2))
* **mqtt:** Publish state for both hass and homie, as the hass device availability_topic is set to the same topic ([#860](https://github.com/Hypfer/Valetudo/issues/860)) ([219bc34](https://github.com/Hypfer/Valetudo/commit/219bc348e95d78ff7dd279c52d498401b5e9d28c))
* **mqtt:** Remove unnecessary availability_topic ([#859](https://github.com/Hypfer/Valetudo/issues/859)) ([aa85205](https://github.com/Hypfer/Valetudo/commit/aa85205fe09007b4b411ef9de754513dda98a4a0)), closes [#858](https://github.com/Hypfer/Valetudo/issues/858)
* **ntpClient:** Support disabling the ntpClient ([949d8e3](https://github.com/Hypfer/Valetudo/commit/949d8e306b5de34c3a3755532f269f009eb50c51)), closes [#925](https://github.com/Hypfer/Valetudo/issues/925)
* **timers:** Copy-paste antipattern ([e331eda](https://github.com/Hypfer/Valetudo/commit/e331eda332e12a138cd1d7e8b8994208f714833a))
* **timers:** Fix zoned scheduled cleanup ([34443b3](https://github.com/Hypfer/Valetudo/commit/34443b3bea4feac5c48844554cd9a816df6f60ff))
* **ui:** DND should be rendered as localtime but stored as UTC ([fd79163](https://github.com/Hypfer/Valetudo/commit/fd79163c3e9328562346a789ad5cec6e73bfe52d))
* **vendor.dreame:** Ignore uploaded multi-map data ([6f3dcc5](https://github.com/Hypfer/Valetudo/commit/6f3dcc5a9a2a39eed6dcc73f2f08d1f438b09cac))
* **vendor.roborock:** Handle DND as UTC ([8fc9ea8](https://github.com/Hypfer/Valetudo/commit/8fc9ea88d81505259f53e0dad5be3ed0e05d5d6f))
* **vendor.roborock:** Roborock only accepts int coordinates ([c7efd6a](https://github.com/Hypfer/Valetudo/commit/c7efd6af1379de2e94594c440c126b74a36a7962))
* Print error message when failing to load config file ([#914](https://github.com/Hypfer/Valetudo/issues/914)) ([9ebda7e](https://github.com/Hypfer/Valetudo/commit/9ebda7eb52185c5d8474344f0e25a653003c1d87))
* **vendor.roborock:** Remove invalid RoborockCombinedVirtualRestrictionsCapability from V1 robot ([5a2bae4](https://github.com/Hypfer/Valetudo/commit/5a2bae404dfebf5e641a3e490e50f7bd254f633b))
* **vendor.viomi:** Fix system timezone in init script ([#876](https://github.com/Hypfer/Valetudo/issues/876)) ([908d5dd](https://github.com/Hypfer/Valetudo/commit/908d5ddc16b03dbe92e1aafe9e0d4f755312e3d4))
* **vendor.viomi:** Syntax error in init script ([2d53b7c](https://github.com/Hypfer/Valetudo/commit/2d53b7cf8a4e25764795859f88f6b4bb24a291e1))
* **vendor.viomi:** Viomi minor bugfixes ([#820](https://github.com/Hypfer/Valetudo/issues/820)) ([4742214](https://github.com/Hypfer/Valetudo/commit/47422140a77e4253606e1e0f360eaaaebb6831f7))

<hr /><em>This discussion was created from the release <a href='https://github.com/Hypfer/Valetudo/releases/tag/2021.05.0'>Valetudo 2021.05.0</a>.</em>

**Comment by AdriaanDeVos** [comment]
Can you also provide binaries as done with the previous releases?

**Reply by Hypfer** [reply]
Please check again. The workflow which automatically generates these has become a bit slower since we're also `upx --ultra-brute` compressing them now. It takes around 10 minutes after the release for the binaries to become available

Here's the build progress:
https://github.com/Hypfer/Valetudo/runs/2674422116

**Reply by AdriaanDeVos** [reply]
Thanks for the release Hypfer. It seems I was just too impatient to test it out! It looks good now :heavy_check_mark:

**Comment by dmlemos** [comment]
> vendor.roborock: Add support for ordered segment cleanup with multiple iterations

Thanks for this feature :)
Is this already included in the UI, or through API support only?

**Reply by Hypfer** [reply]
API only

**Comment by air-fuel-ratio** [comment]
How do i upgrade to 2021.04 prior to this? the instructions https://valetudo.cloud/pages/general/upgrading.html advise to use dustbuilder but they only provide the latest available version, im currently on 2021.02.

**Reply by Hypfer** [reply]
Huh. Indeed

Coming from 2021.02, you should be able to just replace the binary using scp
